src/nspawn/nspawn.c

```diff
@@ -5494,6 +5500,18 @@ static int run(int argc, char *argv[]) {
         * the child. Functions like copy_devnodes() change the umask temporarily. */
         umask(0022);
 
+#if HAVE_SELINUX
+        /* Tell the kernel to create all content for the container with the
+           user specified label */
+        if (arg_selinux_apifs_context)
+                if (setfscreatecon(arg_selinux_apifs_context) < 0)
```

Comment:
this should be arg_selinux_context, not arg_selinux_apifs_context, no?

Reply:
It is correct, this call tells the kernel to create all files created by nspawn to be created with the specified file context.  Specifically this fixes device files created for /dev inside the container.  Potentially other files created.

I did have a bug in the error message.

arg_selinux_context is for processes while arg_selnux_apifs_context is for file system objects.